extensions/amp-story/0.1/amp-story-consent.js

```diff
@@ -214,9 +214,12 @@ export class AmpStoryConsent extends AMP.BaseElement {
       createShadowRootWithStyle(this.element, this.storyConsentEl_, CSS);
 
       // Allow <amp-consent> actions in STAMP (defaults to no actions allowed).
-      this.actions_.addToWhitelist('AMP-CONSENT.accept');
-      this.actions_.addToWhitelist('AMP-CONSENT.prompt');
-      this.actions_.addToWhitelist('AMP-CONSENT.reject');
+      this.actions_.addToWhitelist(
+          {tagOrTarget: 'AMP-CONSENT', method: 'accept'});
+      this.actions_.addToWhitelist(
+          {tagOrTarget: 'AMP-CONSENT', method: 'prompt'});
+      this.actions_.addToWhitelist(
+          {tagOrTarget: 'AMP-CONSENT', method: 'reject'});
```

Comment:
Ah sorry, forgot about these. One of our main runtime projects is to allow minification of public props, which includes objects like these. This means that different binaries (v0.js and extensions) will minify `tagOrTarget` to different names.

To prevent these types of bugs, we can either (1) use string keys and `@dict` type, (2) add the typedef to `amp.extern.js`, or (3) avoid objects in function params i.e.`addToWhitelist(tagOrTarget, method)`.

I'd prefer (3) here since it'll make the method a bit more usable as well. `setWhitelist()` is also a problem but we can probably change it to `clearWhitelist()` given current usage. Let me know if this makes sense.

Reply:
I see, thanks for catching this! Done.

In other projects we use method parameter destructuring pretty often to document the semantics of the arguments (similar to named arguments in other languages):

Without destructuring:

```JS
method('foo', 'bar'); // what are foo and bar?
// Forced to use inline comments to document arguments, which lead to maintenance issues
method(/* firstName= */ 'foo', /* lastName= */ 'bar');
```

With destructuring:

```JS
method({firstName: 'foo', lastName: 'bar'});

function method({firstName, lastName}) {
  // use `firstName` and `lastName` as if they were parameters of the function
}
```

The code becomes self-documenting. But it sounds like this will be a problem with APIs implemented in `v0.js` but are called by extensions... I guess that's just the nature of having separate binaries/compilation units. :(

Reply:
Ah, interesting use of destructuring. Reminds me of Objective-C.